app/src/main/java/ayds/lisboa/songinfo/home/view/SongDescriptionHelperImpl.kt

```diff
@@ -18,8 +21,14 @@ internal class SongDescriptionHelperImpl : SongDescriptionHelper {
                 }\n" +
                         "Artist: ${song.artistName}\n" +
                         "Album: ${song.albumName}\n" +
-                        "Year: ${song.year}"
+                        "Release Date: ${createDate(song)}"
             else -> "Song not found"
         }
     }
+
+    private fun createDate(song: SpotifySong): String{
+        val releaseDatePrecision = song.releaseDatePrecision
+        val releaseDate = song.releaseDate
+        return releaseDateCreator.createDate(releaseDatePrecision,releaseDate)
+    }
```

Comment:
Para mantener le nivel de abstracción, se puede hacer:
```suggestion
    private fun SpotifySong.createDate(): String{
        val releaseDatePrecision = this.releaseDatePrecision
        val releaseDate = this.releaseDate
        return releaseDateCreator.createDate(releaseDatePrecision,releaseDate)
    }
```